tsdb/engine/tsm1/compact_test.go

```diff
@@ -2258,43 +2259,59 @@ func TestDefaultPlanner_PlanOptimize_NoLevel4(t *testing.T) {
 	)
 
 	expFiles := []tsm1.FileStat{}
-	tsm, pLen := cp.PlanOptimize()
+	tsm, pLen, gLen := cp.PlanOptimize()
 	if exp, got := len(expFiles), len(tsm); got != exp {
 		t.Fatalf("tsm file length mismatch: got %v, exp %v", got, exp)
 	} else if pLen != int64(len(tsm)) {
 		t.Fatalf("tsm file plan length mismatch: got %v, exp %v", pLen, exp)
+	} else if gLen != int64(3) {
+		t.Fatalf("generation len plan mismatch: got %v, exp %v", gLen, 3)
 	}
 }
 
-func TestDefaultPlanner_PlanOptimize_Level4(t *testing.T) {
+// This test is added to acount for many TSM files within a group being over 2 GB
+// we want to ensure that the shard will be planned.
+func TestDefaultPlanner_PlanOptimize_LargeMultiGeneration(t *testing.T) {
 	data := []tsm1.FileStat{
 		{
-			Path: "01-04.tsm1",
-			Size: 251 * 1024 * 1024,
+			Path: "01-05.tsm1",
+			Size: 2048 * 1024 * 1024,
 		},
 		{
-			Path: "02-04.tsm1",
-			Size: 1 * 1024 * 1024,
+			Path: "01-06.tsm1",
+			Size: 2048 * 1024 * 1024,
 		},
 		{
-			Path: "03-04.tsm1",
-			Size: 1 * 1024 * 1024,
+			Path: "01-07.tsm1",
+			Size: 2048 * 1024 * 1024,
 		},
 		{
-			Path: "04-04.tsm1",
-			Size: 1 * 1024 * 1024,
+			Path: "01-08.tsm1",
+			Size: 1048 * 1024 * 1024,
 		},
 		{
-			Path: "05-03.tsm1",
-			Size: 2 * 1024 * 1024 * 1024,
+			Path: "02-05.tsm1",
+			Size: 2048 * 1024 * 1024,
 		},
 		{
-			Path: "06-04.tsm1",
-			Size: 2 * 1024 * 1024 * 1024,
+			Path: "02-06.tsm1",
+			Size: 2048 * 1024 * 1024,
 		},
 		{
-			Path: "07-03.tsm1",
-			Size: 2 * 1024 * 1024 * 1024,
+			Path: "02-07.tsm1",
+			Size: 2048 * 1024 * 1024,
+		},
+		{
+			Path: "02-08.tsm1",
+			Size: 1048 * 1024 * 1024,
+		},
+		{
+			Path: "03-04.tsm1",
+			Size: 2048 * 1024 * 1024,
+		},
+		{
+			Path: "03-05.tsm1",
```

Comment:
It would be good to also have this test with the small file as the first one in the generation, as well as the last.